microcosm_eventsource/stores/rollup.py

```diff
@@ -161,17 +161,24 @@ def _container_subquery(self, query):
         """
         Wrap a container query so that it can be used in an aggregation.
 
-        This operation generates as subquery and explicitly maps the subquery to the table name so that ordering
-        can be applied in `_rollup_query` without modifying the container store. This operation further wraps the
-        subquery in an alias to the container type so that it can be referenced in `_rollup_query` as an entity.
+        If the container class is polymorphic, then the operation reduces columns - in case
+        of redundant names (e.g. joined table inheritance), otherwise operation explicitly maps
+        the subquery to the table name so that ordering can be applied in `_rollup_query` without modifying
+        the container store. This operation further wraps the subquery in an alias
+        to the container type so that it can be referenced in `_rollup_query` as an entity.
 
         """
-        return aliased(
-            self.container_type,
-            query.subquery(
+        if self._is_polymorphic():
+            subquery = query.subquery(
+                reduce_columns=True,
+                with_labels=True,
+            )
+        else:
+            subquery = query.subquery(
```

Comment:
question: do we actually need LN178 (`self.container_type.__tablename__`)?

Reply:
I assume so it was already there. IIRC, it has something to do with how the table name gets identified in the roll up clause, which is alluded to in the doc string.